### ISSUE TITLE
[BUG] - dialog actions dont work when editing a record

### ISSUE THREAD

**Issue by ZiggyQubert** [premieroctet/next-admin#422]
### Description

When editing a record for a model that has a dialog action defined the action does not work and a `TypeError: open is not a function` error is thrown

### Reproduction URL

n/a

### Reproduction steps

```bash
1. Create a dialog action for a model
2. edit a record in that model
3. select the created action
```

### Next router

App router

### Next Admin version

6.0.0

### Next Admin options

```ts
actions: [
    {
      id: 'exportAsJson',
      title: 'Export Json',
      type: 'dialog',
      component: <ExportComponent />,
    },
  ]
```

### Logs

```bash
Unhandled Runtime Error
TypeError: open is not a function

Call Stack
onClick
node_modules\@premieroctet\next-admin\dist\components\ActionDropdownItem.js (18:1)
handleEvent
node_modules\@radix-ui\primitive\dist\index.mjs (4:1)
React
HTMLUnknownElement.callCallback
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (20565:1)
Object.invokeGuardedCallbackImpl
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (20614:1)
invokeGuardedCallback
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (20689:1)
invokeGuardedCallbackAndCatchFirstError
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (20703:1)
executeDispatch
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (32128:1)
processDispatchQueueItemsInOrder
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (32160:1)
processDispatchQueue
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (32173:1)
dispatchEventsForPlugins
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (32184:1)
eval
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (32374:1)
batchedUpdates$1
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (24953:1)
batchedUpdates
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (28844:1)
dispatchEventForPluginEventSystem
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (32373:1)
dispatchEvent
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (30141:1)
dispatchDiscreteEvent
node_modules\next\dist\compiled\react-dom\cjs\react-dom.development.js (30112:1)
```

### Browsers

Chrome

**Comment by cregourd** [comment]
Hi, thank you for opening a file. Please let us know how you are using this action feature. Do you need a data object? Do you need a data object with all the relationships included? What do you use/display in your dialog box? 

There is currently an issue with using of the action dialog in the form, and we're looking for the best solution. In the meantime, you seem to want to use a JSON export, for that, you can still use :

-  "classic" action:  https://github.com/premieroctet/next-admin/blob/a20b10f40afe5ec8f3c26baa05fad1ac6b29e4d5/apps/example/options.tsx#L157-L165
- export system (list only) : https://github.com/premieroctet/next-admin/blob/a20b10f40afe5ec8f3c26baa05fad1ac6b29e4d5/apps/example/options.tsx#L179-L183

We're working on this patch 🚀

**Comment by ZiggyQubert** [comment]
Thanks.

The action I'm creating is to export only the current/selected records (dialog actions also dont seam to show in the upper action menu on the list view when items are selected) for data.

As for on the edit page the same data that is passed in when calling from the list view would be a good start, ideally the starting data and the current data (if fields have ben edited) would be good.

I'm also using the exports as you have mentioned. (note - having the url field able to be a function that returns a url would also be great)

**Comment by cregourd** [comment]
Hello,

We have just released [v6.0.1](https://www.npmjs.com/package/@premieroctet/next-admin/v/6.0.1). 

There is a fix for this problem, you can now use the dialog action in lists and forms. Your dialog component can access an array of the involved objects, where an object has its own scalars and those of its relations (2-deep relationship). In the case of a form action, this component will get an array containing a single element (the current object).

Thanks for the issue, and please let us know what you think 🚀.
